jetty-websocket/websocket-jetty-common/src/main/java/org/eclipse/jetty/websocket/common/JettyWebSocketFrameHandler.java

```diff
@@ -365,16 +351,37 @@ private void onPingFrame(Frame frame, Callback callback)
             {
                 throw new WebSocketException(endpointInstance.getClass().getSimpleName() + " PING method error: " + cause.getMessage(), cause);
             }
+
+            callback.succeeded();
+            demand();
         }
         else
         {
-            // Automatically respond
-            ByteBuffer payload = BufferUtil.copy(frame.getPayload());
-            getSession().getRemote().sendPong(payload, WriteCallback.NOOP);
+            // Automatically respond if output is open.
+            if (coreSession.isOutputOpen())
```

Comment:
Same concern about checking for the output in an input method (`onPingFrame()`).